conans/model/new_build_info.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import os
+from collections import OrderedDict
+
+from conans.errors import ConanException
+from conans.model.build_info import DefaultOrderedDict
+
+_DIRS_VAR_NAMES = ["includedirs", "srcdirs", "libdirs", "resdirs", "bindirs", "builddirs",
+                  "frameworkdirs"]
+_FIELD_VAR_NAMES = ["system_libs", "frameworks", "libs", "defines", "cflags", "cxxflags",
+                   "sharedlinkflags", "exelinkflags"]
+
+
+class _BaseNewCppInfo(object):
+
+    def __init__(self):
+        self.names = {}
+        self.build_modules = {}
+
+        # ###### DIRECTORIES
+        self.includedirs = []  # Ordered list of include paths
+        self.srcdirs = []  # Ordered list of source paths
+        self.libdirs = []  # Directories to find libraries
+        self.resdirs = []  # Directories to find resources, data, etc
+        self.bindirs = []  # Directories to find executables and shared libs
+        self.builddirs = []
+        self.frameworkdirs = []
+
+        # ##### FIELDS
+        self.system_libs = []  # Ordered list of system libraries
+        self.frameworks = []  # Macos .framework
+        self.libs = []  # The libs to link against
+        self.defines = []  # preprocessor definitions
+        self.cflags = []  # pure C flags
+        self.cxxflags = []  # C++ compilation flags
+        self.sharedlinkflags = []  # linker flags
+        self.exelinkflags = []  # linker flags
+
+        self.sysroot = ""
+        self.requires = []
+
+        # ##### PATTERNS
+        self.include_patterns = []
+        self.lib_patterns = []
+        self.bin_patterns = []
+        self.src_patterns = []
+        self.build_patterns = []
+        self.res_patterns = []
+        self.framework_patterns = []
+
+    def get_name(self, generator):
+        return self.names.get(generator)
+
+    def to_absolute_paths(self, root_folder):
+        for name in _DIRS_VAR_NAMES:
+            setattr(self, name, [os.path.join(root_folder, c) for c in getattr(self, name)])
+
+        self.build_modules = {k: [os.path.join(root_folder, e) for e in v]
+                                  for k, v in self.build_modules.items()}
+
+
+class NewCppInfo(_BaseNewCppInfo):
```

Comment:
Maybe it would simplify the design a bit to assume that there is always at least 1 component, without a name, or with a canonical name like the package name, and forward all the base definitions to that component, and start managing everything as a component?

```python
# assuming we use ``None`` as internal identifier of the package component
self.cpp_info.includedirs = ["myinc"]  # equal self.cpp_info.components[None].includedirs = ["myinc"]
```
Just an idea to consider.

Reply:
Interesting. I will check it.

Reply:
Done. check